src/client/opamAuxCommands.ml

```diff
@@ -293,27 +293,37 @@ let autopin_aux st ?quiet ?(for_view=false) ?recurse ?subpath ?locked
           (* check of the target to avoid repin of pin to update with `opam
              install .` and loose edited opams *)
           let pinned_pkg = OpamPinned.package st nf.pin_name in
-          OpamSwitchState.primary_url st pinned_pkg = Some nf.pin.pin_url
-          &&
-          (match OpamSwitchState.opam_opt st pinned_pkg with
-           | Some opam ->
-             (match locked, OpamFile.OPAM.locked opam with
-              | Some ext , Some ext' -> String.equal ext ext'
-              | None, None -> true
-              | _ -> false)
-           | None -> false)
-          &&
-          (* For `opam show`, we need to check does the opam file changed to
-             perform a simulated pin if so *)
-          (not for_view ||
-           match
-             OpamSwitchState.opam_opt st pinned_pkg,
-             OpamFile.OPAM.read_opt nf.pin.pin_file
-           with
-           | Some opam0, Some opam ->
-             let opam = OpamFile.OPAM.with_locked_opt nf.pin.pin_locked opam in
-             OpamFile.OPAM.equal opam0 opam
-           | _, _ -> false)
+          match OpamSwitchState.opam_opt st pinned_pkg with
+          | Some opam0 ->
+            (match OpamFile.OPAM.read_opt nf.pin.pin_file with
+             | Some opam ->
+               let opam = OpamFile.OPAM.with_locked_opt nf.pin.pin_locked opam in
```

Comment:
Commit 449436922834aa09e89efa58153a48c9fd72ab5c lost in the PR split

Reply:
It was discarded on purpose as to me https://github.com/ocaml/opam/pull/6438 removes the actual change made to the opam definition that we have to think about. But i've readded it, i don't mind either way.